src/main/java/io/cryostat/agent/ConfigModule.java

```diff
@@ -159,6 +165,64 @@ public static Config provideConfig() {
         }
     }
 
+    @Provides
+    @Singleton
+    @Named(CRYOSTAT_AGENT_TRUSTSTORES)
+    public static List<TruststoreConfig> provideTruststoreConfigs(Config config) {
+        Map<Integer, TruststoreConfig> truststoreMap = new HashMap<>();
+
+        List<String> propertyNames =
+                StreamSupport.stream(config.getPropertyNames().spliterator(), false)
+                        .filter(e -> e.startsWith(CRYOSTAT_AGENT_TRUSTSTORES))
+                        .collect(Collectors.toList());
+
+        for (String name : propertyNames) {
+
+            String[] parts = name.split("\\.");
+            if (parts.length < 3) {
+                log.error(
+                        "Invalid truststore config property name format: {}. Rename to"
+                                + " 'truststore.cert[CERT_NUMBER].CERT_PROPERTY",
+                        name);
+                continue;
+            }
+
+            int truststoreNumber = 0;
+            try {
+                truststoreNumber = Integer.parseInt(parts[1].substring(5, 6));
+            } catch (IllegalArgumentException e) {
+                log.error(
+                        "Invalid truststore config property name format: {}. Make sure"
+                                + " the certificate number is valid",
+                        name);
```

Comment:
I think there should be a `continue` here for similar reasons as the previous one.